src/webapi/rendering_context.rs

```diff
@@ -30,6 +46,201 @@ impl RenderingContext for CanvasRenderingContext2d {
+        }
+    }
+
+    /// Adds a cubic Bézier curve to the path. It requires three points. The first two points 
+    /// are control points and the third one is the end point. The starting point is the last 
+    /// point in the current path, which can be changed using moveTo() before creating the Bézier curve.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/bezierCurveTo)
+    pub fn bezier_curve_to(&self, cp1x: f64, cp1y: f64, cp2x: f64, cp2y: f64, x: f64, y: f64) {
+        js! { @(no_return)
+            @{&self.0}.bezierCurveTo(@{cp1x}, @{cp1y}, @{cp2x}, @{cp2y}, @{x}, @{y});
+        }
+    }
+
+    /// Sets all pixels in the rectangle defined by starting point (x, y) and size (width, height) 
+    /// to transparent black, erasing any previously drawn content.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/clearRect)
+    pub fn clear_rect(&self, x: f64, y: f64, width: f64, height: f64) {
+        js! { @(no_return)
+            @{&self.0}.clearRect(@{x}, @{y}, @{width}, @{width}, @{height});
+        }
+    }
+
+    /// Turns the path currently being built into the current clipping path.
+    /// ctx.clip(path, fillRule) is not supported because [(Path2D)](https://developer.mozilla.org/en-US/docs/Web/API/Path2D) is still experimental
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/clip)
+    pub fn clip(&self, fill_rule: Option<FillRule>) {
+        if let Some(fill_rule) = fill_rule {
+            let fill_rule_str;
+            match fill_rule {
+                FillRule::NonZero => {
+                    fill_rule_str = "nonzero";
+                }
+
+                FillRule::EvenOdd => {
+                    fill_rule_str = "evenodd";
+                }
+            }
+            js! { @(no_return)
+                @{&self.0}.clip(@{fill_rule_str});
+            }    
+        }
+        else {
+            js! { @(no_return)
+                @{&self.0}.clip();
+            }
+        }
+    }
+
+    /// Causes the point of the pen to move back to the start of the current sub-path. It tries 
+    /// to add a straight line (but does not actually draw it) from the current point to the start. 
+    /// If the shape has already been closed or has only one point, this function does nothing.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/closePath)
+    pub fn close_path(&self) {
+        js! { @(no_return)
+            @{&self.0}.closePath();
+        }
+    }
+
+    /// Creates a gradient along the line given by the coordinates represented by the parameters.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createLinearGradient)
+    pub fn create_linear_gradient(&self, x0: f64, y0: f64, x1: f64, y1: f64) {
```

Comment:
You don't have to implement every type, but in case a function returns something (like a `CanvasGradient` in this case) please create a dummy private type and return it. (Just so that we won't have to change the prototype in the future.)